tensorboard/components/tf_storage/storage.ts

```diff
@@ -223,9 +223,13 @@ function readComponent(): string {
 /**
  * Write component to URI.
  */
-function writeComponent(component: string) {
+function writeComponent(component: string, useLocationReplace = false) {
   if (tf_globals.useHash()) {
-    window.location.hash = component;
+      if (useLocationReplace) {
+          window.location.replace(window.location.origin + window.location.pathname + '#' + component);
+      } else {
```

Comment:
nit: After `pathname`, could you concatenate `window.location.search`?

Reply:
Actually, you can just do: `window.location.replace('#' + component);`

Any preceding parts of the replacement URL not given will resolve against the current URL, including the query string.  This is the same reason why you can use `<a href="#top">Go to Top</a>` and similar things inline in a page.